### PR TITLE
Add octopus build link to teamcity output

### PR DESCRIPTION
**Pull request by taliesins**
Added a link from GUI to Octopus deployments as well
Add octopus build link to teamcity output

This feature was requested #35

### CODE REVIEW

File: src/OctopusPuppet.Cmd/Program.cs [hunk 1]
```diff
@@ -190,16 +194,17 @@ private static IEnumerable<IComponentVertexDeployer> GetDeployers(string targetE
...
             var environmentDeployment = LoadEnvironmentDeploy(opts.EnvironmentDeploymentPath);
             var deployers = GetDeployers(opts.TargetEnvironment, true, true, opts.OctopusUrl, opts.OctopusApiKey);
-            return Deploy(notifier, environmentDeployment, opts.MaximumParallelDeployments, deployers);
+            return Deploy(notifier, logMessager, environmentDeployment, opts.MaximumParallelDeployments, deployers);
         }
 
         private static int CommandLineParsingError(IEnumerable<Error> errors)
         {
-            var notifier = GetNotifier(false, false);
+            var notifier = GetNotifier(null, false, false);
```
Comment (margani) [inline src/OctopusPuppet.Cmd/Program.cs]: Can we change this so we have GetNotifier() by having parameters with default values?

Comment (taliesins) [reply src/OctopusPuppet.Cmd/Program.cs]: Much cleaner now :>

Comment (margani) [reply src/OctopusPuppet.Cmd/Program.cs]: Good! Can we do it for other parameters as well?

File: src/OctopusPuppet/LogMessager/ILogMessager.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+﻿using OctopusPuppet.Scheduler;
+
+namespace OctopusPuppet.LogMessager
+{
+    public interface ILogMessager
```
Comment (slawomir-brzezinski-at-interxion) [inline src/OctopusPuppet/LogMessager/ILogMessager.cs]: Must say that `ILogMessager` is too close in meaning to something that delivers the messages (that's what it is in real life) and there can very well be a component in this code that does have that responsibility.

Maybe just `ILogMessages`, `ILogMessageWordings`, `ICommonDeploymentMessageWordings`? Or maybe you can think of something even better?

File: src/OctopusPuppet.Gui/Views/NavigationService.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+using System.Text.RegularExpressions;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Documents;
+
+namespace OctopusPuppet.Gui.Views
+{
+    public static class NavigationService
+    {
+        // Copied from http://geekswithblogs.net/casualjim/archive/2005/12/01/61722.aspx
```
Comment (slawomir-brzezinski-at-interxion) [inline src/OctopusPuppet.Gui/Views/NavigationService.cs]: Is this still true?

File: src/OctopusPuppet.Gui/Views/NavigationService.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Diagnostics;
+using System.Text.RegularExpressions;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Documents;
+
+namespace OctopusPuppet.Gui.Views
+{
+    public static class NavigationService
```
Comment (slawomir-brzezinski-at-interxion) [inline src/OctopusPuppet.Gui/Views/NavigationService.cs]: A really vague name.

Maybe `URLLinkifier`?

File: src/OctopusPuppet.Gui/Views/NavigationService.cs (hunk 3 above)
Comment (taliesins) [reply src/OctopusPuppet.Gui/Views/NavigationService.cs]: We changed the regex, the rest is the same

File: src/OctopusPuppet/LogMessager/ILogMessager.cs (hunk 2 above)
Comment (taliesins) [reply src/OctopusPuppet/LogMessager/ILogMessager.cs]: I was having trouble with a name to call the class. I like ILogMessages

File: src/OctopusPuppet.OctopusProvider/OctopusLogMessager.cs [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        }
+
+        private string GetOctopusDeploymentUrl(ComponentDeploymentVertex componentDeploymentVertex)
+        {
+            if (componentDeploymentVertex == null || string.IsNullOrEmpty(componentDeploymentVertex.DeploymentId))
+            {
+                return null;
+            }
+            
+            var deploymentUri = string.Format("{0}/app#/deployments/{1}", _url, componentDeploymentVertex.DeploymentId);
```
Comment (slawomir-brzezinski-at-interxion) [inline src/OctopusPuppet.OctopusProvider/OctopusLogMessager.cs]: Are we able to step into 21th century and use C# 6.0 string interpolation? ;)

File: src/OctopusPuppet.Gui/Views/NavigationService.cs (hunk 4 above)
Comment (taliesins) [reply src/OctopusPuppet.Gui/Views/NavigationService.cs]: Done

File: src/OctopusPuppet/LogMessager/ILogMessager.cs (hunk 2 above)
Comment (ma499) [reply src/OctopusPuppet/LogMessager/ILogMessager.cs]: I'm also struggling with the name - maybe ILogMessageGenerator or ILogMessageTemplates.

But I'm more wondering why this is an interface? Looks more like a helper/utility class with one and only one implementation.

**Pull request merged**